File: fccalua/game/static.lua
require 'i18n'

--
-- Goods
-- 
goods = {}
local function good(name, can_buy, initial_prices, raw_material)
   local g = {
      name = name,
      can_buy = can_buy,
      initial_prices = initial_prices,
      raw_material = raw_material,
   }
   table.insert(goods, g)
   return g
end

food =      good(_'Food', true, { 1,8 })
cotton =    good(_'Cotton', true, { 4,6 })
barley =    good(_'Barley', true, { 4,6 })
olives =    good(_'Olives', true, { 6,8 })
wood =      good(_'Wood', true, { 1,6 })
copper =    good(_'Copper', true, { 5,8 })
mud =       good(_'Mud', true, { 1,5 })
tin =       good(_'Tin', true, { 7,9 })
gold =      good(_'Gold', true, { 15,16 })
clothes =   good(_'Clothes', true, { 12,13 }, { cotton })
beer =      good(_'Beer', true, { 10,11 }, { beer })
olive_oil = good(_'Olive Oil', true, { 12,13 }, { olives })
furniture = good(_'Furniture', true, { 10,11 }, { wood })
utensils =  good(_'Utensils', true, { 12,13 }, { copper })
bricks =    good(_'Bricks', true, { 1,8 }, { mud })
riding =    good(_'Riding Animals', true, { 5,8 })
bronze =    good(_'Bronze', true, { 9,10 }, { copper, tin })
weapons =   good(_'Weapons', true, { 6,7 }, { wood, bronze })
scrolls =   good(_'Scrolls', false)
prayers =   good(_'Prayers', false)

-- 
-- Terrains
--
terrains = {}
local function terrain(name, cost_to_enter, defensive_bonus, production, switch_status)
local function good_production(p)
   if not p then return {} end
   local pp = {}
   for i,v in ipairs(p) do pp[goods[i]] = v end
   return pp
end
local t = {
   name = name,
   production = good_production(production),
   switch_status = switch_status,
}
if switch_status then switch_status.switch_status = t end -- swap
table.insert(terrains, t)
return t
end

--                 Name              MoveCost      Barley      Mud
--                                      Defence       Olives      Tin
--                                           Food        Wood        Gold
--                                              Cotton      Copper
tundra   = terrain(_'Tundra',        1, 0, { 2, 0, 0, 0, 0, 2, 2, 1, 0 })
desert   = terrain(_'Desert',        1, 0, { 1, 0, 0, 2, 0, 1, 3, 2, 0 })
plains   = terrain(_'Planis',        1, 0, { 4, 2, 1, 0, 0, 1, 1, 0, 0 })
prairie  = terrain(_'Prairie',       1, 0, { 3, 3, 0, 0, 0, 0, 0, 0, 0 })
steppe   = terrain(_'Steppe',        1, 0, { 2, 2, 3, 0, 0, 0, 0, 0, 0 })
marsh    = terrain(_'Marsh',         2, 1, { 3, 1, 2, 0, 0, 2, 4, 2, 0 })

boreal_f = terrain(_'Boreal forest', 2, 2, { 1, 0, 0, 0, 2, 1, 1, 1, 0 }, tundra)
scrub    = terrain(_'Scrubland',     1, 2, { 1, 0, 0, 4, 1, 1, 2, 1, 0 }, desert)
mixed_f  = terrain(_'Mixed forest',  2, 2, { 2, 1, 0, 0, 3, 0, 1, 0, 0 }, plains)
savannah = terrain(_'Savannah',      1, 1, { 1, 1, 0, 0, 2, 0, 1, 0, 0 }, prairie)
woodland = terrain(_'Woodland',      2, 2, { 1, 0, 1, 0, 2, 0, 1, 0, 0 }, steppe)
swamp    = terrain(_'Swamp',         3, 3, { 2, 0, 0, 0, 1, 1, 3, 2, 0 }, marsh)

arctic   = terrain(_'Arctic',        2, 0, { 0, 0, 0, 0, 0, 1, 0, 1, 1 })
sea      = terrain(_'Sea',           1, 0, { 3, 0, 0, 0, 0, 0, 0, 0, 0 })
ocean    = terrain(_'Ocean',         1, 0, { 3, 0, 0, 0, 0, 0, 0, 0, 0 })
hills    = terrain(_'Hills',         2, 4, { 1, 0, 0, 0, 0, 3, 0, 3, 0 })
mountain = terrain(_'Mountains',     3, 6, { 0, 0, 0, 0, 0, 3, 0, 3, 1 })

--
-- Military classes
--
militaries = {}
local function military(name, ship, moves, attack, defense, cargo)
local m = {
   name = name,
   ship = ship,
   moves = moves,
   attack = attack,
   defense = defense,
   cargo = cargo,
}
table.insert(militaries, m)
return m
end

peasant         = military(_'Peasant',                 false, 1, 0, 1, 0)
colonist        = military(_'Colonist',                false, 1, 0, 1, 0)
warrior         = military(_'Warrior',                 false, 1, 2, 2, 0)
nomad           = military(_'Nomad',                   false, 4, 1, 1, 0)
chariot         = military(_'Chariot',                 false, 4, 3, 3, 0)
pikeman         = military(_'Pikeman',                 false, 1, 3, 3, 0)
mtd_pikeman     = military(_'Mounted Pikeman',         false, 4, 4, 4, 0)
armored_warrior = military(_'Armed Warrior',           false, 1, 4, 4, 0)
cataphract      = military(_'Cataphract',              false, 4, 5, 5, 0)
catapult        = military(_'Catapult',                false, 1, 7, 5, 0)

clan_member     = military(_'Clan Member',             false, 1, 1, 1, 0)
clan_armed      = military(_'Armed Clan Member',       false, 1, 2, 2, 0)
clan_mounted    = military(_'Clan Member Rider',       false, 4, 2, 2, 0)
clan_mtd_armed  = military(_'Armed Clan Member Armed', false, 4, 3, 3, 0)

elite_soldier   = military(_'Elite Soldier',           false, 1, 4, 4, 0)
elite_chariot   = military(_'Elite Chariot',           false, 4, 5, 5, 0)
faraoh_soldier  = military(_"Faraoh's Guard",          false, 1, 5, 5, 0)
faraoh_chariot  = military(_"Faraoh's Chariot Guard",  false, 4, 6, 6, 0)

penteconter     = military(_'Penteconter',             true,  4, 0, 2, 2)
merchantman     = military(_'Merchantman',             true,  6, 0, 6, 5)
monoreme        = military(_'Monoreme',                true,  4, 1, 3, 0)
bireme          = military(_'Bireme',                  true,  5, 3, 5, 0)
trireme         = military(_'Trireme',                 true,  6, 4, 6, 0)
galley          = military(_'Galley',                  true,  5, 7, 8, 6)
pirate_ship     = military(_'Pirate Ship',             true,  8, 4, 8, 2)


-- 
-- Jobs
--
local function job(name)
   return {
      name = name,
   }
end

farmer       = job(_'Farmer')
cotton_pl    = job(_'Cotton Planter')
barley_pl    = job(_'Barley Planter')
olive_pl     = job(_'Olives Planter')
lumberjack   = job(_'Lumberjack')
copper_miner = job(_'Copper Miner')
mudder       = job(_'Mudder')
tin_miner    = job(_'Tin Miner')
prospector   = job(_'Prospector')
weaver       = job(_'Weaver')
brewer       = job(_'Brewer')
oil_presser  = job(_'Oil Presser')
carpenter    = job(_'Carpenter')
coppersmith  = job(_'Coppersmith')
brickmaker   = job(_'Brickmaker')
metallurgic  = job(_'Metallurgic')
weaponsmith  = job(_'Weaponsmith')
scribe       = job(_'Scribe')
priest       = job(_'Priest')

--
-- Buildings
--
local function building(name, job, raw_good, good, multiplier, cost, 
      copper, min_colony)
   local b = {
      name = name,
      job = job,
      raw_good = raw_good,
      good = good,
      multiplier = mutiplier or 1, -- 3 = factory double production
      cost = cost or 0, -- 0 = starting building
      copper = copper or 0,
      min_colony = min_colony or 0,
   }
   if job then 
      job.building = b
      job.raw_good = raw_good
      job.good = good
   end
   return b
end

weaver_1    = building(_"Weaver's House", weaver, cotton, clothes)
weaver_2    = building(_"Weaver's Shop", weaver, cotton, clothes, 2, 64)a
weaver_3    = building(_"Weaver's Factory", weaver, 
brewer_1    = building(_"
brewer_2    = building(_"
brewer_3    = building(_"
oil_1       = building(_"
oil_2       = building(_"
oil_3       = building(_"
carpenter_1 = building(_"
carpenter_2 = building(_"
copper_1    = building(_"
copper_2    = building(_"
copper_3    = building(_"
brick_1     = building(_"
brick_2     = building(_"
brick_3     = building(_"
stable      = building(_"
bronze_1    = building(_"
bronze_2    = building(_"
bronze_3    = building(_"
scribe_1    = building(_"
scribe_2    = building(_"
scribe_3    = building(_"
temple_1    = building(_"
temple_2    = building(_"
temple_3    = building(_"
embassy     = building(_"
docks_1     = building(_"
docks_2     = building(_"
docks_3     = building(_"
wall_1      = building(_"
wall_2      = building(_"
wall_3      = building(_"
library     = building(_"
oracle      = building(_"
warehouse_1 = building(_"
warehouse_2 = building(_"
palace      = building(_"
monument    = building(_"

File: fccaruby/cociv/static.rb
require 'i18n'

class Static

  @@id = 0

  attr_reader :id

  def initialize
    @id = @@id
    @@id += 1
  end

  def inspect
    return "<##{self.class.name}:#{@name}>"
  end

  # The methods below allow that if the classes are loaded
  # twice, they remaind the same when compared or used as
  # keys in hashes. This is specially useful in marshalling
  # (serialization).

  def eql?(o)
    return false if not o.is_a? Static
    return @id == o.id
  end

  def ==(o)
    return false if not o.is_a? Static
    return @id == o.id
  end

  def hash
    @id
  end

end

# 
# Directions
DIRECTIONS = {
  1 => [-1, 1], 2 => [ 0, 1], 3 => [ 1, 1],
  4 => [-1, 0],               6 => [ 1, 0],
  7 => [-1,-1], 8 => [ 0,-1], 9 => [ 1,-1]
}

#
# Goods
# 
class Good < Static

  @@all = []

  attr_reader :name, :raw, :can_buy, :initial_prices, :raw_material, :all

  def initialize(name, raw, can_buy, initial_prices=nil, raw_material=nil)
    super()
    @name, @raw, @can_buy, @initial_prices, @raw_material = name, raw, can_buy, initial_prices, raw_material
    @@all << self
  end

  def Good.all; @@all; end

end

Food =      Good.new(_('Food'),      true,  true, [  1, 8 ])
Cotton =    Good.new(_('Cotton'),    true,  true, [  4, 6 ])
Barley =    Good.new(_('Barley'),    true,  true, [  4, 6 ])
Olives =    Good.new(_('Olives'),    true,  true, [  6, 8 ])
Wood =      Good.new(_('Wood'),      true,  true, [  1, 6 ])
Copper =    Good.new(_('Copper'),    true,  true, [  5, 8 ])
Mud =       Good.new(_('Mud'),       true,  true, [  1, 5 ])
Tin =       Good.new(_('Tin'),       true,  true, [  7, 9 ])
Gold =      Good.new(_('Gold'),      true,  true, [ 15,16 ])
Clothes =   Good.new(_('Clothes'),   false, true, [ 12,13 ], [ Cotton ])
Beer =      Good.new(_('Beer'),      false, true, [ 10,11 ], [ Barley ])
Olive_oil = Good.new(_('Olive Oil'), false, true, [ 12,13 ], [ Olives ])
Furniture = Good.new(_('Furniture'), false, true, [ 10,11 ], [ Wood ])
Utensils =  Good.new(_('Utensils'),  false, true, [ 12,13 ], [ Copper ])
Bricks =    Good.new(_('Bricks'),    false, true, [  1, 8 ], [ Mud ])
Riding =    Good.new(_('Animals'),   false, true, [  5, 8 ])
Bronze =    Good.new(_('Bronze'),    false, true, [  9,10 ], [ Copper, Tin ])
Weapons =   Good.new(_('Weapons'),   false, true, [  6, 7 ], [ Wood, Bronze ])
Scrolls =   Good.new(_('Scrolls'),   false, false)
Prayers =   Good.new(_('Prayers'),   false, false)


#
# Military classes
#
class Military < Static

  @@all = []

  attr_reader :name, :ship, :moves, :attack, :defense, :cargo

  def initialize(name, ship, moves, attack, defense, cargo)
    super()
    @name, @ship, @moves, @attack, @defense, @cargo = name, ship, moves, attack, defense, cargo
    @@all << self
  end

end

Peasant         = Military.new(_('Peasant'),                 false, 1, 0, 1, 0)
Colonist        = Military.new(_('Colonist'),                false, 1, 0, 1, 0)
Caravan         = Military.new(_('Caravan'),                 false, 2, 0, 1, 2)
Warrior         = Military.new(_('Warrior'),                 false, 1, 2, 2, 0)
Nomad           = Military.new(_('Nomad'),                   false, 4, 1, 1, 0)
Chariot         = Military.new(_('Chariot'),                 false, 4, 3, 3, 0)
Pikeman         = Military.new(_('Pikeman'),                 false, 1, 3, 3, 0)
Mtd_pikeman     = Military.new(_('Mounted Pikeman'),         false, 4, 4, 4, 0)
Armored_warrior = Military.new(_('Armed Warrior'),           false, 1, 4, 4, 0)
Cataphract      = Military.new(_('Cataphract'),              false, 4, 5, 5, 0)
Catapult        = Military.new(_('Catapult'),                false, 1, 7, 5, 0)

Clan_member     = Military.new(_('Clan Member'),             false, 1, 1, 1, 0)
Clan_armed      = Military.new(_('Armed Clan Member'),       false, 1, 2, 2, 0)
Clan_mounted    = Military.new(_('Clan Member Rider'),       false, 4, 2, 2, 0)
Clan_mtd_armed  = Military.new(_('Armed Clan Member Armed'), false, 4, 3, 3, 0)

Elite_soldier   = Military.new(_('Elite Soldier'),           false, 1, 4, 4, 0)
Elite_chariot   = Military.new(_('Elite Chariot'),           false, 4, 5, 5, 0)
Faraoh_soldier  = Military.new(_("Faraoh's Guard"),          false, 1, 5, 5, 0)
Faraoh_chariot  = Military.new(_("Faraoh's Chariot Guard"),  false, 4, 6, 6, 0)

Penteconter     = Military.new(_('Penteconter'),             true,  4, 0, 2, 2)
Merchantman     = Military.new(_('Merchantman'),             true,  6, 0, 6, 5)
Monoreme        = Military.new(_('Monoreme'),                true,  4, 1, 3, 0)
Bireme          = Military.new(_('Bireme'),                  true,  5, 3, 5, 0)
Trireme         = Military.new(_('Trireme'),                 true,  6, 4, 6, 0)
Galley          = Military.new(_('Galley'),                  true,  5, 7, 8, 6)
Pirate_ship     = Military.new(_('Pirate Ship'),             true,  8, 4, 8, 2)


# 
# Jobs
#
class Job < Static

  @@all = []

  attr_reader :name, :raw
  attr_accessor :building, :raw_good, :good

  def initialize(name, raw=false, good=nil)
    super()
    @name = name
    @raw = raw
    @good = good
    @@all << self
  end

  def Job.all; @@all; end

end

Farmer       = Job.new(_('Farmer'), true, Food)
Fisherman    = Job.new(_('Fisherman'), true, Food)
Cotton_pl    = Job.new(_('Cotton Planter'), true, Cotton)
Barley_pl    = Job.new(_('Barley Planter'), true, Barley)
Olive_pl     = Job.new(_('Olives Planter'), true, Olives)
Lumberjack   = Job.new(_('Lumberjack'), true, Wood)
Copper_miner = Job.new(_('Copper Miner'), true, Copper)
Mudder       = Job.new(_('Mudder'), true, Mud)
Tin_miner    = Job.new(_('Tin Miner'), true, Tin)
Prospector   = Job.new(_('Prospector'), true, Gold)
Weaver       = Job.new(_('Weaver'))
Brewer       = Job.new(_('Brewer'))
Oil_presser  = Job.new(_('Oil Presser'))
Carpenter    = Job.new(_('Carpenter'))
Coppersmith  = Job.new(_('Coppersmith'))
Potter       = Job.new(_('Potter'))
Metallurgic  = Job.new(_('Metallurgic'))
Weaponsmith  = Job.new(_('Weaponsmith'))
Scribe       = Job.new(_('Scribe'))
Priest       = Job.new(_('Priest'))


# 
# Terrains
#
class Terrain < Static

  @@all = []

  attr_reader :all, :name, :ovl, :cost_to_enter, :defensive_bonus, :production, :suggested_job
  attr_accessor :switch_status

  def initialize(name, ovl, cost_to_enter, defensive_bonus, production, 
                 suggested_job, switch_status=nil)
    super()
    @name, @ovl, @cost_to_enter, @defensive_bonus, @switch_status, @suggested_job =
      name, ovl, cost_to_enter, defensive_bonus, switch_status, suggested_job
    @production = {}
    i = 0
    Good.all.each do |good|
      @production[good] = production[i]
      i += 1
    end
    @switch_status.switch_status = self if @switch_status
    @@all << self
  end

end

#                      Name                    MoveCost       Barley      Mud
#                                           Ovl*   Defence       Olives      Tin
#                                                       Food        Wood        Gold
#                                                          Cotton      Copper
Tundra   = Terrain.new(_('Tundra'),          1, 0, 0, [ 2, 0, 0, 0, 0, 2, 2, 1, 0 ], Copper_miner)
Desert   = Terrain.new(_('Desert'),         13, 1, 0, [ 1, 0, 0, 2, 0, 1, 3, 2, 0 ], Mudder)
Plains   = Terrain.new(_('Planis'),         14, 1, 0, [ 4, 2, 1, 0, 0, 1, 1, 0, 0 ], Farmer)
Prairie  = Terrain.new(_('Prairie'),        15, 1, 0, [ 3, 3, 0, 0, 0, 0, 0, 0, 0 ], Cotton_pl)
Steppe   = Terrain.new(_('Steppe'),         16, 1, 0, [ 2, 2, 3, 0, 0, 0, 0, 0, 0 ], Barley_pl)
Marsh    = Terrain.new(_('Marsh'),          17, 2, 1, [ 3, 1, 2, 0, 0, 2, 4, 2, 0 ], Mudder)

Boreal_f = Terrain.new(_('Boreal forest'),   6, 2, 2, [ 1, 0, 0, 0, 2, 1, 1, 1, 0 ], Lumberjack, Tundra)
Scrub    = Terrain.new(_('Scrubland'),       7, 1, 2, [ 1, 0, 0, 4, 1, 1, 2, 1, 0 ], Olive_pl, Desert)
Mixed_f  = Terrain.new(_('Mixed forest'),    8, 2, 2, [ 2, 1, 0, 0, 3, 0, 1, 0, 0 ], Lumberjack, Plains)
Savannah = Terrain.new(_('Savannah'),        9, 1, 1, [ 1, 1, 0, 0, 2, 0, 1, 0, 0 ], Lumberjack, Prairie)
Woodland = Terrain.new(_('Woodland'),       10, 2, 2, [ 1, 0, 1, 0, 2, 0, 1, 0, 0 ], Lumberjack, Steppe)
Swamp    = Terrain.new(_('Swamp'),          11, 3, 3, [ 2, 0, 0, 0, 1, 1, 3, 2, 0 ], Mudder, Marsh)

Arctic   = Terrain.new(_('Arctic'),          5, 2, 0, [ 0, 0, 0, 0, 0, 1, 0, 1, 1 ], Prospector)
Sea      = Terrain.new(_('Sea'),             2, 1, 0, [ 3, 0, 0, 0, 0, 0, 0, 0, 0 ], Fisherman)
Ocean    = Terrain.new(_('Ocean'),           1, 1, 0, [ 3, 0, 0, 0, 0, 0, 0, 0, 0 ], Fisherman)
Hills    = Terrain.new(_('Hills'),           4, 2, 4, [ 1, 0, 0, 0, 0, 3, 0, 3, 0 ], Tin_miner)
Mountain = Terrain.new(_('Mountains'),       3, 3, 6, [ 0, 0, 0, 0, 0, 3, 0, 3, 1 ], Copper_miner)
# *Ovl: Overlap for terrains.


#
# Buildings
#
class BuildingType < Static

  @@all = []

  attr_reader :name, :max_units, :job, :raw_good, :good, :multiplier, :cost, :Copper, 
              :min_colony

  def initialize(name, max_units, job, raw_good, good, multiplier=1, cost=0, copper=0, 
                 min_colony=0)
    super()
    @name, @max_units, @job, @raw_good, @good, @multiplier, @cost, @copper, @min_colony = 
      name, max_units, job, raw_good, good, multiplier, cost, copper, min_colony
    if @job
      @job.building = self
      @job.raw_good = raw_good
      @job.good = good
    end
    @@all << self
  end

  def BuildingType.all; @@all; end

end

Weaver_1    = BuildingType.new(_("Weaver's House"), 2, Weaver, Cotton, Clothes)
Weaver_2    = BuildingType.new(_("Weaver's Shop"), 3, Weaver, Cotton, Clothes, 2, 64, 20)
Weaver_3    = BuildingType.new(_("Cloth Mill"), 3, Weaver, Cotton, Clothes, 3, 160, 100, 8)
Brewer_1    = BuildingType.new(_("Brewer's House"), 2, Brewer, Barley, Beer)
Brewer_2    = BuildingType.new(_("Brewer's Shop"), 3, Brewer, Barley, Beer, 2, 64, 20)
Brewer_3    = BuildingType.new(_("Beer Mill"), 3, Brewer, Barley, Beer, 3, 160, 100, 8)
Oil_1       = BuildingType.new(_("Small Oil Press"), 2, Oil_presser, Olives, Olive_oil)
Oil_2       = BuildingType.new(_("Big Oil Press"), 3, Oil_presser, Olives, Olive_oil, 2, 64, 20)
Oil_3       = BuildingType.new(_("Olive Oil Mill"), 3, Oil_presser, Olives, Olive_oil, 3, 160, 100, 8)
Carpenter_1 = BuildingType.new(_("Carpenter's House"), 2, Carpenter, Wood, Furniture)
Carpenter_2 = BuildingType.new(_("Carpenter's Shop"), 3, Carpenter, Wood, Furniture, 2, 64, 20)
Carpenter_3 = BuildingType.new(_("Lumber Mill"), 3, Carpenter, Wood, Furniture, 3, 160, 100, 8)
Copper_1    = BuildingType.new(_("Coppersmith's House"), 2, Coppersmith, Copper, Utensils)
Copper_2    = BuildingType.new(_("Coppersmith's Shop"), 3, Coppersmith, Copper, Utensils, 2, 64, 20)
Copper_3    = BuildingType.new(_("Utensils Factory"), 3, Coppersmith, Copper, Utensils, 3, 160, 100, 8)
Brick_1     = BuildingType.new(_("Small Klin"), 2, Potter, Mud, Bricks)
Brick_2     = BuildingType.new(_("Big Klin"), 3, Potter, Mud, Bricks, 2, 64, 20)
Brick_3     = BuildingType.new(_("Brick Factory"), 3, Potter, Mud, Bricks, 3, 160, 100, 8)
Stable      = BuildingType.new(_("Stable"), 0, nil, Food, Riding, 2, 64, 20)
Bronze_1    = BuildingType.new(_("Metallurgic's Shop"), 2, Metallurgic, [ Copper, Tin ], Bronze)
Bronze_2    = BuildingType.new(_("Metalworks"), 3, Metallurgic, [ Copper, Tin ], Bronze, 2, 64, 20)
Bronze_3    = BuildingType.new(_("Machine Shop"), 3, Metallurgic, [ Copper, Tin ], Bronze, 3, 160, 100, 8)
Scribe_1    = BuildingType.new(_("Tablet House"), 2, Scribe, nil, Scrolls)
Scribe_2    = BuildingType.new(_("Gymnasium"), 3, Scribe, nil, Scrolls, 2, 64, 20)
Scribe_3    = BuildingType.new(_("Academy"), 3, Scribe, nil, Scrolls, 3, 160, 100, 8)
Temple_1    = BuildingType.new(_("Worship Yard"), 1, Priest, nil, Prayers, 1, 80)
Temple_2    = BuildingType.new(_("Shrine"), 3, Priest, nil, Prayers, 2, 64, 20)
Temple_3    = BuildingType.new(_("Temple"), 3, Priest, nil, Prayers, 3, 160, 100, 8)
Embassy     = BuildingType.new(_("Embassy"), 0, nil, nil, nil, 3, 160, 100, 8)
Docks_1     = BuildingType.new(_("Docks"), 0, nil, nil, nil, 1, 52)
Docks_2     = BuildingType.new(_("Drydock"), 0, nil, nil, nil, 1, 80, 50, 4)
Docks_3     = BuildingType.new(_("Shipyard"), 0, nil, nil, nil, 1, 240, 100, 8)
Wall_1      = BuildingType.new(_("Wooden Wall"), 0, nil, nil, nil, 1, 64)
Wall_2      = BuildingType.new(_("Stone Wall"), 0, nil, nil, nil, 2, 120, 100)
Wall_3      = BuildingType.new(_("Masonry Wall"), 0, nil, nil, nil, 3, 240, 200)
Library     = BuildingType.new(_("Library"), 0, nil, nil, nil, 0, 120, 20)
Oracle      = BuildingType.new(_("Oracle"), 0, nil, nil, nil, 0, 120, 20)
Warehouse_1 = BuildingType.new(_("Warehouse"), 0, nil, nil, nil, 2, 80)
Warehouse_2 = BuildingType.new(_("Big Warehouse"), 0, nil, nil, nil, 3, 120, 20)
Palace      = BuildingType.new(_("Palace"), 0, nil, nil, nil, 0, 240, 200)
Monument    = BuildingType.new(_("Monument"), 0, nil, nil, nil, 0, 320, 300)

InitialBuildings = [ Weaver_1, Brewer_1, Oil_1, Carpenter_1, Copper_1, Brick_1, Bronze_1 ]

# Nation
Israel = nil
